File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int l = 0;
        int r = 1;
        int maxP = 0;
        while(r < prices.length){
            var profit = prices[r] - prices[l];
            if(prices[l] >= prices[r]){
                l = r;
            }else if(prices[l] < prices[r]){
                maxP = Math.max(maxP, profit);
            }
            r++;
        }
        return maxP;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0, profit = 0, l = 0, r = 1, n = prices.size();
        while(r < n){
            profit = prices[r]-prices[l];
            if(prices[l] >= prices[r]){
                l = r;
            }else if(prices[l] < prices[r]){
                max_profit = max(profit, max_profit);
            }
            r++;
        }
        return max_profit;
    }
};